clang/include/clang/Sema/Sema.h

```diff
@@ -13435,6 +13435,13 @@ class Sema final : public SemaBase {
     return CodeSynthesisContexts.size() > NonInstantiationEntries;
   }
 
+  using EntityPrinter = llvm::function_ref<void(llvm::raw_ostream &)>;
+
+  /// \brief create a Requirement::SubstitutionDiagnostic with only a
+  /// SubstitutedEntity and DiagLoc using Sema's allocator.
```

Comment:
nit: Probably this could be more precise: we're in fact constructing the object on ASTContext, so maybe we can say "using ASTContext's allocator"?